internal/codeintel/codenav/internal/lsifstore/lsifstore_documents.go

```
@@ -23,37 +24,64 @@ func (s *store) SCIPDocument(ctx context.Context, uploadID int, path core.Upload
+		return core.None[*scip.Document](), err
+	}
+	if doc, ok := documents[path]; ok {
+		return core.Some(doc), nil
+	} else {
+		return core.None[*scip.Document](), nil
+	}
+}
+
+func (s *store) SCIPDocuments(ctx context.Context, uploadID int, paths []core.UploadRelPath) (_ map[core.UploadRelPath]*scip.Document, err error) {
+	stringPaths := genslices.Map(paths, func(p core.UploadRelPath) string { return p.RawValue() })
+	ctx, _, endObservation := s.operations.scipDocuments.With(ctx, &err, observation.Args{Attrs: []attribute.KeyValue{
+		attribute.Int("uploadID", uploadID),
+		attribute.StringSlice("paths", stringPaths),
+	}})
+	defer endObservation(1, observation.Args{})
+
+	scanner := basestore.NewMapScanner(func(dbs dbutil.Scanner) (core.UploadRelPath, *scip.Document, error) {
 		var compressedSCIPPayload []byte
-		if err := dbs.Scan(&compressedSCIPPayload); err != nil {
-			return nil, err
+		var path string
+		emptyPath := core.NewUploadRelPathUnchecked("")
+		if err := dbs.Scan(&path, &compressedSCIPPayload); err != nil {
+			return emptyPath, nil, err
 		}
 
 		scipPayload, err := shared.Decompressor.Decompress(bytes.NewReader(compressedSCIPPayload))
 		if err != nil {
-			return nil, err
+			return emptyPath, nil, err
 		}
 
 		var document scip.Document
 		if err := proto.Unmarshal(scipPayload, &document); err != nil {
-			return nil, err
+			return emptyPath, nil, err
 		}
-		return &document, nil
+		return core.NewUploadRelPathUnchecked(path), &document, nil
 	})
-	doc, ok, err := scanner(s.db.Query(ctx, sqlf.Sprintf(fetchSCIPDocumentQuery, uploadID, path.RawValue())))
-	if err != nil || !ok {
-		return core.None[*scip.Document](), err
+	searchPaths := make([]*sqlf.Query, 0, len(paths))
+	for _, path := range stringPaths {
+		searchPaths = append(searchPaths, sqlf.Sprintf("%s", path))
+	}
+	docs, err := scanner(s.db.Query(ctx, sqlf.Sprintf(fetchSCIPDocumentsQuery, uploadID, sqlf.Join(searchPaths, ","))))
+	if err != nil {
+		return nil, nil
```

Hm, does this swallow the error? Or does assignment to out variable `err` prevent that? I don't know what Go does in this case..

---

Thanks for catching that 🤦